File: exercises/week05/exercise-dp-two/src/composite/TestCompositePattern.java
package composite;

public class TestCompositePattern {
    public static void main(String[] args) {
        HtmlTag parentTag = new HtmlParentElement("<html>");
        parentTag.setStartTag("<html>");
        parentTag.setEndTag("</html>");
        HtmlTag p1 = new HtmlParentElement("<body>");
        p1.setStartTag("<body>");
        p1.setEndTag("</body>");
        parentTag.addChildTag(p1);
        HtmlTag child1 = new HtmlElement("<P>");
        child1.setStartTag("<P>");
        child1.setEndTag("</P>");
        child1.setTagBody("Testing html tag library");
        p1.addChildTag(child1);
        child1 = new HtmlElement("<P>");
        child1.setStartTag("<P>");
        child1.setEndTag("</P>");
        child1.setTagBody("Paragraph 2");
        p1.addChildTag(child1);
        parentTag.generateHtml();
    }
}

File: exercises/week05/exercise-dp-two-scala/src/composite/TestCompositePattern.scala
package composite

object TestCompositePattern {
  def main(args: Array[String]) {
    val parentTag: HtmlTag = new HtmlParentElement("<html>")
    parentTag.setStartTag("<html>")
    parentTag.setEndTag("</html>")
    val p1: HtmlTag = new HtmlParentElement("<body>")
    p1.setStartTag("<body>")
    p1.setEndTag("</body>")
    parentTag.addChildTag(p1)
    var child1: HtmlTag = new HtmlElement("<P>")
    child1.setStartTag("<P>")
    child1.setEndTag("</P>")
    child1.setTagBody("Testing html tag library")
    p1.addChildTag(child1)
    child1 = new HtmlElement("<P>")
    child1.setStartTag("<P>")
    child1.setEndTag("</P>")
    child1.setTagBody("Paragraph 2")
    p1.addChildTag(child1)
    parentTag.generateHtml
  }
}